spec/models/manageiq/providers/openshift/container_manager/refresher_spec.rb

```diff
@@ -127,6 +127,228 @@ def normal_refresh
+                     :match_requests_on => [:path,], :record => :new_episodes) do
+
+      EmsRefresh.refresh(@ems)
+
+      assert_projects_created
+      assert_table_counts_before_deletions
+
+      delete_objects
+    end
+
+    VCR.use_cassette("#{described_class.name.underscore}_after_openshift_deletions",
+                     :match_requests_on => [:path,], :record => :new_episodes) do
+
+      EmsRefresh.refresh(@ems)
+
+      assert_table_counts_after_deletions
+      assert_specific_objects_disconnected
+    end
+  end
+
+  def delete_objects
+    @ems.connect_client("v1", "delete_project").delete_project("my-project-0")
+
+    %w(pod service route resourceQuota limitRange persistentVolumeClaim template).each do |entity|
+      method_name = "delete_#{entity.underscore}"
+      object_name = "my-#{entity.downcase}"
+      @ems.connect_client("v1", method_name).send(method_name, object_name, "my-project-1")
+    end
+
+    delete_route_labels
+    delete_template_parameters
+  end
+
+  # updating specific route with no labels will delete its current labels
+  def delete_route_labels
+    updated_route = Kubeclient::Route.new(
+      :metadata => {
+        :name      => "my-route",
+        :namespace => "my-project-2"
+      },
+      :spec     => {
+        :to => {
+          :kind => "Service",
+          :name => "service-name"
+        }
+      }
+    )
+    @ems.connect_client("v1", "update_route").update_route(updated_route)
+  end
+
+  # updating specific template with no parameters will delete its current parameters
+  def delete_template_parameters
+    template = ContainerTemplate.find_by(:name => "my-template", :container_project => @project2)
+    updated_template = Kubeclient::Template.new(
+      :metadata => {
+        :name            => "my-template",
+        :namespace       => "my-project-2",
+        :resourceVersion => template.resource_version
+      },
+      :objects  => template.objects
+    )
+    @ems.connect_client("v1", "update_template").update_template(updated_template)
+  end
+
+  def assert_projects_created
+    @project0 = ContainerProject.find_by(:name => "my-project-0")
+    @project1 = ContainerProject.find_by(:name => "my-project-1")
+    @project2 = ContainerProject.find_by(:name => "my-project-2")
+
+    [@project0, @project1, @project2].each do |project|
+      expect(project).not_to be_nil
+      expect(project.ext_management_system).to eq(@ems)
+    end
+  end
+
+  def assert_specific_objects_disconnected
+    assert_specific_projects_disconnected
+    assert_specific_pods_disconnected
+    assert_specific_container_definitions_disconnected
+    assert_specific_containers_disconnected
+  end
+
+  def assert_specific_projects_disconnected
+    @project0.reload
+    expect(@project0).not_to be_nil
+    expect(@project0.ext_management_system).to be_nil
+    expect(@project0.deleted_on).not_to be_nil
+    expect(@project0.old_ems_id).to eq(@ems.id)
```

Comment:
Above 4 lines will change with https://github.com/ManageIQ/manageiq/pull/15251
They repeat for several objects here, so I'd extract them to an `assert_disconnected(@project0)` helper.